qemu/tests/qemu_guest_agent.py

```diff
@@ -721,6 +720,41 @@ def _action_after_fsfreeze(self, *args):
             raise error.TestFail("FS freeze failed, guest still can"
                                  " write file")
 
+    def gagent_check_reset_reboot_shutdown(self, test, params, env):
+
+        """
+        Execute "system_reset" and send "shutdown,reboot" command to guest agent after FS freezed
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        vm = env.get_vm(params["main_vm"])
+        if not vm.is_alive():
+            test.skip("VM died,can't continue the test loop.")
+        self.gagent.fsfreeze()
+        self.gagent.verify_fsfreeze_status(self.gagent.FSFREEZE_STATUS_FROZEN)
+
+        try:
+            self.gagent.shutdown(self.gagent.SHUTDOWN_MODE_POWERDOWN)
+        except Exception:
+            test.fail("guest-shutdown command returned a wrong message")
+        else:
+            logging.info("guest FS is still freeze status")
+        self.gagent.verify_fsfreeze_status(self.gagent.FSFREEZE_STATUS_FROZEN)
+
+        try:
```

Comment:
removed try.  use 'if' to instead of try.
        if self.gagent.shutdown(self.gagent.SHUTDOWN_MODE_POWERDOWN):
            self.gagent.verify_fsfreeze_status(self.gagent.FSFREEZE_STATUS_FROZEN)

        if self.gagent.shutdown(self.gagent.SHUTDOWN_MODE_REBOOT):
            self.gagent.verify_fsfreeze_status(self.gagent.FSFREEZE_STATUS_FROZEN)

        if vm.monitor.cmd("system_reset"):
            self.gagent.verify_fsfreeze_status(self.gagent.FSFREEZE_STATUS_THAWED)